File: Vista/TicketDialog.cs
using Microsoft.Maui.Controls;
using Microsoft.Maui.Graphics;
using Microsoft.Maui.Layouts;
using System;
using System.Linq;
using System.Threading.Tasks;
using punto_de_venta_C_.Modelo;

namespace punto_de_venta_C_.Vista
{
    public class TicketDialog : ContentPage
    {
        private readonly Venta venta;
        private readonly ConfiguracionTienda config;
        private readonly double cambio;
        private readonly double descuento;
        private readonly double efectivoRecibido;
        private TaskCompletionSource<bool> tcs = new TaskCompletionSource<bool>();

        public TicketDialog(Venta venta, ConfiguracionTienda config, double cambio, double efectivoRecibido)
            : this(venta, config, cambio, venta.getDescuento(), efectivoRecibido)
        {
        }

        public TicketDialog(Venta venta, ConfiguracionTienda config, double cambio, double descuento, double efectivoRecibido)
        {
            this.venta = venta;
            this.config = config;
            this.cambio = cambio;
            this.efectivoRecibido = efectivoRecibido;
            this.descuento = venta.getDescuento() > 0 ? venta.getDescuento() : descuento;
            Title = "Ticket de Venta";
            BackgroundColor = Estilos.BG_BLANCO;
            construir();
        }

        private void construir()
        {
            var absoluteLayout = new AbsoluteLayout
            {
                WidthRequest = 380,
                HeightRequest = 660,
                BackgroundColor = Estilos.BG_BLANCO
            };

            int y = 0;

            var barra = new BoxView
            {
                Color = Estilos.INDIGO,
                HeightRequest = 8
            };
            AbsoluteLayout.SetLayoutBounds(barra, new Rect(0, 0, 380, 8));
            AbsoluteLayout.SetLayoutFlags(barra, AbsoluteLayoutFlags.None);
            absoluteLayout.Add(barra);
            y = 18;

            absoluteLayout.Add(centrado(config.getNombre(), 15, FontAttributes.Bold, y));
            y += 22;

            var lblSuc = centrado(config.getSucursal(), Estilos.FUENTE_XS_SIZE, FontAttributes.None, y);
            lblSuc.TextColor = Estilos.TEXTO_SECUNDARIO;
            absoluteLayout.Add(lblSuc);
            y += 16;

            var lblRfc = centrado("RFC: " + config.getRfc(), Estilos.FUENTE_XS_SIZE, FontAttributes.None, y);
            lblRfc.TextColor = Estilos.TEXTO_SECUNDARIO;
            absoluteLayout.Add(lblRfc);
            y += 16;

            absoluteLayout.Add(sep(y));
            y += 14;

            absoluteLayout.Add(centrado(venta.getFecha().ToString("dd/MM/yyyy HH:mm:ss"), Estilos.FUENTE_XS_SIZE, FontAttributes.None, y));
            y += 16;
            absoluteLayout.Add(centrado("Cajero: " + venta.getCajero().ToUpper(), Estilos.FUENTE_XS_SIZE, FontAttributes.None, y));
            y += 16;

            var lblNum = centrado("Ticket #" + venta.getId(), Estilos.FUENTE_XS_SIZE, FontAttributes.None, y);
            lblNum.TextColor = Estilos.TEXTO_TENUE;
            absoluteLayout.Add(lblNum);
            y += 16;

            absoluteLayout.Add(sep(y));
            y += 14;

            // Items List
            var itemsLayout = new VerticalStackLayout();
            foreach (var item in venta.getItems())
            {
                var rowGrid = new Grid
                {
                    HeightRequest = 34,
                    ColumnDefinitions = new ColumnDefinitionCollection
                    {
                        new ColumnDefinition { Width = GridLength.Star },
                        new ColumnDefinition { Width = 110 }
                    }
                };

                var itemTitleLayout = new VerticalStackLayout();
                var codLbl = new Label
                {
                    Text = "[" + item.getProducto().getId() + "] " + item.getProducto().getNombre().ToUpper(),
                    FontSize = Estilos.FUENTE_XS_SIZE,
                    TextColor = Estilos.TEXTO_PRINCIPAL
                };
                var detLbl = new Label
                {
                    Text = string.Format("  {0:0.00} x ${1:0.00}", item.getCantidad(), item.getProducto().getPrecio()),
                    FontSize = Estilos.FUENTE_XS_SIZE - 2,
                    TextColor = Estilos.TEXTO_TENUE
                };
                itemTitleLayout.Add(codLbl);
                itemTitleLayout.Add(detLbl);
                rowGrid.Add(itemTitleLayout, 0, 0);

                var subLbl = new Label
                {
                    Text = string.Format("${0:0.00}", item.getSubtotal()),
                    FontSize = Estilos.FUENTE_XS_SIZE,
                    TextColor = Estilos.TEXTO_PRINCIPAL,
                    HorizontalOptions = LayoutOptions.End,
                    VerticalOptions = LayoutOptions.Center
                };
                rowGrid.Add(subLbl, 1, 0);

                itemsLayout.Add(rowGrid);
            }

            var itemsScroll = new ScrollView
            {
                Content = itemsLayout,
                HeightRequest = 180,
                BackgroundColor = Estilos.BG_BLANCO
            };
            AbsoluteLayout.SetLayoutBounds(itemsScroll, new Rect(14, y, 352, 180));
            AbsoluteLayout.SetLayoutFlags(itemsScroll, AbsoluteLayoutFlags.None);
            absoluteLayout.Add(itemsScroll);
            y += 190;

            absoluteLayout.Add(sep(y));
            y += 10;

            double precioTotal = venta.getItems().Sum(i => i.getSubtotal());
            double montoDescuento = descuento;
            double montoAPagar = venta.getTotal();

            fila(absoluteLayout, "Precio total de compra:", string.Format("${0:0.00}", precioTotal), y, Estilos.TEXTO_SECUNDARIO);
            y += 18;

            Color colorDesc = montoDescuento > 0 ? Estilos.EMERALD : Estilos.TEXTO_TENUE;
            fila(absoluteLayout, "Monto de descuento:", string.Format("-${0:0.00}", montoDescuento), y, colorDesc);
            y += 18;

            absoluteLayout.Add(sep(y));
            y += 10;

            filaGrande(absoluteLayout, "MONTO A PAGAR:", string.Format("${0:0.00}", montoAPagar), y, Estilos.TEXTO_PRINCIPAL);
            y += 22;

            absoluteLayout.Add(sep(y));
            y += 10;

            fila(absoluteLayout, "Efectivo recibido:", string.Format("${0:0.00}", efectivoRecibido), y, Estilos.TEXTO_SECUNDARIO);
            y += 18;
            fila(absoluteLayout, "Cambio:", string.Format("${0:0.00}", cambio), y, Estilos.EMERALD);
            y += 20;

            var gracias = centrado("Gracias por su compra", Estilos.FUENTE_SMALL_SIZE, FontAttributes.None, y);
            gracias.TextColor = Estilos.TEXTO_TENUE;
            absoluteLayout.Add(gracias);
            y += 30;

            var btnImprimir = new Button
            {
                Text = "Imprimir / Cerrar",
                FontAttributes = FontAttributes.Bold,
                BackgroundColor = Estilos.BG_OSCURO,
                TextColor = Colors.White,
                HeightRequest = 44,
                WidthRequest = 352
            };
            btnImprimir.Clicked += async (s, e) =>
            {
                await DisplayAlert("Impresión", "Ticket enviado a la cola de impresión.", "OK");
                tcs.SetResult(true);
                await Navigation.PopModalAsync();
            };
            AbsoluteLayout.SetLayoutBounds(btnImprimir, new Rect(14, y + 4, 352, 44));
            AbsoluteLayout.SetLayoutFlags(btnImprimir, AbsoluteLayoutFlags.None);
            absoluteLayout.Add(btnImprimir);

            Content = new Grid
            {
                Children = { absoluteLayout },
                BackgroundColor = Estilos.BG_BLANCO,
                HorizontalOptions = LayoutOptions.Center,
                VerticalOptions = LayoutOptions.Center
            };
        }

        private Label centrado(string texto, double fontSize, FontAttributes fontAttributes, int y)
        {
            var lbl = new Label
            {
                Text = texto,
                FontSize = fontSize,
                FontAttributes = fontAttributes,
                HorizontalTextAlignment = TextAlignment.Center,
                TextColor = Estilos.TEXTO_PRINCIPAL
            };
            AbsoluteLayout.SetLayoutBounds(lbl, new Rect(0, y, 380, 18));
            AbsoluteLayout.SetLayoutFlags(lbl, AbsoluteLayoutFlags.None);
            return lbl;
        }

        private BoxView sep(int y)
        {
            var s = new BoxView
            {
                Color = Estilos.BORDE,
                HeightRequest = 1
            };
            AbsoluteLayout.SetLayoutBounds(s, new Rect(14, y, 352, 1));
            AbsoluteLayout.SetLayoutFlags(s, AbsoluteLayoutFlags.None);
            return s;
        }

        private void fila(AbsoluteLayout p, string label, string valor, int y, Color colorValor)
        {
            var lbl = new Label
            {
                Text = label,
                FontSize = Estilos.FUENTE_XS_SIZE,
                TextColor = Estilos.TEXTO_SECUNDARIO
            };
            AbsoluteLayout.SetLayoutBounds(lbl, new Rect(14, y, 180, 16));
            AbsoluteLayout.SetLayoutFlags(lbl, AbsoluteLayoutFlags.None);
            p.Add(lbl);

            var val = new Label
            {
                Text = valor,
                FontSize = Estilos.FUENTE_XS_SIZE,
                TextColor = colorValor,
                HorizontalTextAlignment = TextAlignment.End
            };
            AbsoluteLayout.SetLayoutBounds(val, new Rect(200, y, 166, 16));
            AbsoluteLayout.SetLayoutFlags(val, AbsoluteLayoutFlags.None);
            p.Add(val);
        }

        private void filaGrande(AbsoluteLayout p, string label, string valor, int y, Color colorValor)
        {
            var lbl = new Label
            {
                Text = label,
                FontSize = Estilos.FUENTE_BOLD_SIZE,
                FontAttributes = FontAttributes.Bold,
                TextColor = Estilos.TEXTO_PRINCIPAL
            };
            AbsoluteLayout.SetLayoutBounds(lbl, new Rect(14, y, 180, 20));
            AbsoluteLayout.SetLayoutFlags(lbl, AbsoluteLayoutFlags.None);
            p.Add(lbl);

            var val = new Label
            {
                Text = valor,
                FontSize = 15,
                FontAttributes = FontAttributes.Bold,
                TextColor = colorValor,
                HorizontalTextAlignment = TextAlignment.End
            };
            AbsoluteLayout.SetLayoutBounds(val, new Rect(180, y, 186, 20));
            AbsoluteLayout.SetLayoutFlags(val, AbsoluteLayoutFlags.None);
            p.Add(val);
        }

        public async Task<bool> mostrar()
        {
            if (Application.Current?.MainPage != null)
            {
                await Application.Current.MainPage.Navigation.PushModalAsync(new NavigationPage(this));
            }
            return await tcs.Task;
        }
    }
}

File: Vista/TicketDialog.py
from PyQt6.QtWidgets import QDialog, QVBoxLayout, QLabel, QFrame, QPushButton, QScrollArea, QWidget
from PyQt6.QtCore import Qt
from PyQt6.QtGui import QFont, QColor
from PyQt6.QtPrintSupport import QPrinter, QPrintDialog
from PyQt6.QtGui import QPainter
from Vista.Estilos import Estilos

class TicketDialog(QDialog):
    def __init__(self, parent, venta, config, cambio, descuento=None, efectivoRecibido=None):
        super().__init__(parent)
        self.setWindowTitle("Ticket de Venta")
        self.setModal(True)
        self.venta = venta
        self.config = config
        self.cambio = cambio
        
        if efectivoRecibido is None:
            self.efectivoRecibido = descuento
            self.descuento = venta.getDescuento() if venta.getDescuento() > 0 else cambio
        else:
            self.efectivoRecibido = efectivoRecibido
            self.descuento = venta.getDescuento() if venta.getDescuento() > 0 else descuento
            
        self.construir()

    def construir(self):
        self.setFixedSize(430, 700)
        
        contenedor = QWidget()
        contenedor.setStyleSheet(f"background-color: {Estilos.BG_BLANCO.name()};")
        
        self.panelTicket = QWidget(contenedor)
        self.panelTicket.setStyleSheet(f"background-color: {Estilos.BG_BLANCO.name()};")
        
        y = 0
        
        barra = QFrame(self.panelTicket)
        barra.setStyleSheet(f"background-color: {Estilos.INDIGO.name()};")
        barra.setGeometry(0, 0, 380, 8)
        y = 18
        
        lblNombre = self.centrado(self.config.getNombre(), QFont("Monospaced", 15, QFont.Weight.Bold), y)
        self.panelTicket.children().append(lblNombre)
        y += 22
        
        lblSuc = self.centrado(self.config.getSucursal(), Estilos.FUENTE_XS, y)
        lblSuc.setStyleSheet(f"color: {Estilos.TEXTO_SECUNDARIO.name()};")
        y += 16
        
        lblRfc = self.centrado(f"RFC: {self.config.getRfc()}", Estilos.FUENTE_XS, y)
        lblRfc.setStyleSheet(f"color: {Estilos.TEXTO_SECUNDARIO.name()};")
        y += 16
        
        self.sep(y)
        y += 14
        
        fecha_str = self.venta.getFecha().strftime("%d/%m/%Y %H:%M:%S") if self.venta.getFecha() else ""
        self.centrado(fecha_str, Estilos.FUENTE_XS, y)
        y += 16
        
        self.centrado(f"Cajero: {self.venta.getCajero().upper()}", Estilos.FUENTE_XS, y)
        y += 16
        
        lblNum = self.centrado(f"Ticket #{self.venta.getId()}", Estilos.FUENTE_XS, y)
        lblNum.setStyleSheet(f"color: {Estilos.TEXTO_TENUE.name()};")
        y += 16
        
        self.sep(y)
        y += 14
        
        for item in self.venta.getItems():
            cod = QLabel(f"[{item.getProducto().getId()}] {item.getProducto().getNombre().upper()}", self.panelTicket)
            cod.setFont(Estilos.FUENTE_XS)
            cod.setGeometry(14, y, 240, 16)
            
            sub = QLabel(f"${item.getSubtotal():.2f}", self.panelTicket)
            sub.setFont(Estilos.FUENTE_XS)
            sub.setAlignment(Qt.AlignmentFlag.AlignRight)
            sub.setGeometry(256, y, 110, 16)
            y += 16
            
            det = QLabel(f"  {item.getCantidad():.2f} x ${item.getProducto().getPrecio():.2f}", self.panelTicket)
            det.setFont(Estilos.FUENTE_XS)
            det.setStyleSheet(f"color: {Estilos.TEXTO_TENUE.name()};")
            det.setGeometry(14, y, 240, 14)
            y += 18
            
        self.sep(y)
        y += 10
        
        precioTotal = sum(item.getSubtotal() for item in self.venta.getItems())
        montoAPagar = self.venta.getTotal()
        
        self.fila("Precio total de compra:", f"${precioTotal:.2f}", y, Estilos.TEXTO_SECUNDARIO)
        y += 18
        
        colorDesc = Estilos.EMERALD if self.descuento > 0 else Estilos.TEXTO_TENUE
        self.fila("Monto de descuento:", f"-${self.descuento:.2f}", y, colorDesc)
        y += 18
        
        self.sep(y)
        y += 10
        
        self.filaGrande("MONTO A PAGAR:", f"${montoAPagar:.2f}", y, Estilos.TEXTO_PRINCIPAL)
        y += 22
        
        self.sep(y)
        y += 10
        
        self.fila("Efectivo recibido:", f"${self.efectivoRecibido:.2f}", y, Estilos.TEXTO_SECUNDARIO)
        y += 18
        
        self.fila("Cambio:", f"${self.cambio:.2f}", y, Estilos.EMERALD)
        y += 20
        
        gracias = self.centrado("Gracias por su compra", Estilos.FUENTE_SMALL, y)
        gracias.setStyleSheet(f"color: {Estilos.TEXTO_TENUE.name()};")
        y += 30
        
        self.panelTicket.setGeometry(0, 0, 380, y)
        
        btnImprimir = QPushButton("Imprimir / Cerrar", contenedor)
        btnImprimir.setFont(Estilos.FUENTE_BOLD)
        btnImprimir.setCursor(Qt.CursorShape.PointingHandCursor)
        btnImprimir.setStyleSheet(f"""
            QPushButton {{
                background-color: {Estilos.BG_OSCURO.name()};
                color: white;
                border: none;
            }}
        """)
        btnImprimir.setGeometry(14, y + 4, 352, 44)
        btnImprimir.clicked.connect(self.imprimir)
        
        contenedor.setMinimumSize(380, y + 56)
        
        scroll = QScrollArea(self)
        scroll.setWidget(contenedor)
        scroll.setStyleSheet("border: none;")
        
        layout = QVBoxLayout(self)
        layout.setContentsMargins(0, 0, 0, 0)
        layout.addWidget(scroll)

    def centrado(self, texto, fuente, y):
        lbl = QLabel(texto, self.panelTicket)
        lbl.setFont(fuente)
        lbl.setAlignment(Qt.AlignmentFlag.AlignCenter)
        lbl.setGeometry(0, y, 380, 18)
        return lbl

    def sep(self, y):
        s = QFrame(self.panelTicket)
        s.setFrameShape(QFrame.Shape.HLine)
        s.setStyleSheet(f"color: {Estilos.BORDE.name()}; background-color: {Estilos.BORDE.name()};")
        s.setGeometry(14, y, 352, 1)
        return s

    def fila(self, label, valor, y, colorValor):
        lbl = QLabel(label, self.panelTicket)
        lbl.setFont(Estilos.FUENTE_XS)
        lbl.setStyleSheet(f"color: {Estilos.TEXTO_SECUNDARIO.name()};")
        lbl.setGeometry(14, y, 180, 16)
        
        val = QLabel(valor, self.panelTicket)
        val.setFont(Estilos.FUENTE_XS)
        val.setStyleSheet(f"color: {colorValor.name()};")
        val.setAlignment(Qt.AlignmentFlag.AlignRight)
        val.setGeometry(200, y, 166, 16)

    def filaGrande(self, label, valor, y, colorValor):
        lbl = QLabel(label, self.panelTicket)
        lbl.setFont(Estilos.FUENTE_BOLD)
        lbl.setStyleSheet(f"color: {Estilos.TEXTO_PRINCIPAL.name()};")
        lbl.setGeometry(14, y, 180, 20)
        
        val = QLabel(valor, self.panelTicket)
        val.setFont(QFont("Monospaced", 15, QFont.Weight.Bold))
        val.setStyleSheet(f"color: {colorValor.name()};")
        val.setAlignment(Qt.AlignmentFlag.AlignRight)
        val.setGeometry(180, y, 186, 20)

    def imprimir(self):
        printer = QPrinter(QPrinter.PrinterMode.HighResolution)
        dialog = QPrintDialog(printer, self)
        if dialog.exec() == QPrintDialog.DialogCode.Accepted:
            painter = QPainter()
            painter.begin(printer)
            self.panelTicket.render(painter)
            painter.end()
        self.accept()

    def mostrar(self):
        self.exec()
